File: SourceCode/Python/Binary/checkBST.py
// Assumes that the tree is not allowed to contain duplicates
# /* *** ODSATag: checkBST *** */
def checkBST(node, low, high):
    if node is None: return True # Empty subtree
    rootval = node.value()
    if rootval <= low || rootval >= high:
        return False # Out of range
    if not checkBST(node.left(), low, rootval):
        return False # Left side failed
    return checkBST(node.right(), rootval, high)
# /* *** ODSAendTag: checkBST *** */

File: SourceCode/Java_Generic/Binary/checkBST.java
// Assumes that the tree is not allowed to contain duplicates
/* *** ODSATag: checkBST *** */
static <E extends Comparable<E>> boolean checkBST(BinNode<E> node, E low, E high) {
    if (node == null) return true;  // Empty subtree

    E rootval = node.value();
    if (rootval.compareTo(low) <= 0 || rootval.compareTo(high) >= 0)
        return false;  // Out of range

    return checkBST(node.left(), low, rootval)    // Check left subtree
        && checkBST(node.right(), rootval, high); // Check right subtree
}
/* *** ODSAendTag: checkBST *** */
